clients/dataset/dataset.go

```diff
@@ -83,60 +43,42 @@ var _ error = ErrInvalidDatasetAPIResponse{}
 
 // Client is a dataset api client which can be used to make requests to the server
 type Client struct {
-	cli           RHTTPClient
-	ctxcli        RCHTTPClient
-	url           string
-	internalToken string
+	cli common.RCHTTPClienter
+	url string
 }
 
-// New creates a new instance of Client with a given dataset api url
-func New(datasetAPIURL string) *Client {
+// NewAPIClient creates a new instance of Client with a given dataset api url and the relevant tokens
+func NewAPIClient(datasetAPIURL, serviceToken, xDownloadServiceToken, florenceToken string) *Client {
```

Comment:
I am not familiar with the xDownloadServiceToken - is this something other than the service token?

Reply:
yes, it's essentially a token specific to the download service (why it's not using the usual `Authentication:` header pattern, I don't recall: Matt will know)